types don't match module format

Ran into a weird issue with this module after switching typescript to the 'node16' module format.
It boils down to the code is exporting a function as a cjs module, but the index.d.ts definitions are written as if it's esm.

error TS2349: This expression is not callable.
  Type 'typeof import(".../node_modules/pg-tsquery/index")' has no call signatures.


I think the fix would be something like this in index.d.ts instead of the export default:
```typescript
declare function tsquery(options?: TsqueryOptions): (str: string) => string;
export = tsquery
```

For now a work around is this goofy thing:
```typescript
import { default as Tsquery } from 'pg-tsquery'

type TsQueryFunc = (options?: Tsquery.TsqueryOptions) => (str: string) => string
const tsqueryParser = (Tsquery as unknown as TsQueryFunc)()
```

The typescript guys built this little tool that seems to agree with my analysis:
https://arethetypeswrong.github.io/?p=pg-tsquery%408.4.1

--- a/.gitignore
+++ b/.gitignore
@@ -2,3 +2,4 @@ node_modules
 package-lock.json
 .nyc_output
 .vscode
+.idea

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,4 +1,5 @@
-export interface TsqueryOptions {
+declare namespace tsquery {
+  export interface TsqueryOptions {
     word?: RegExp; // regex for a word. It must contain named captures: word (required), quote (optional), phrase (optional), negated (optional)
     negated?: RegExp; // regex to detect if the expression following an operator is negated, this is useful for example with 'foo!bar', you can parse it as foo&!bar by adding the negation as part of and
     quotedWordSep?: RegExp; // regex for word delimiters inside quotes
@@ -10,9 +11,9 @@ export interface TsqueryOptions {
     prefix?: RegExp; // regex for detecting `prefix` operator (placed at the end of a word to match words starting like it)
     tailOp?: string; // default operator to use with tail (unparsed suffix of the query, if any)
     singleQuoteReplacement?: string; // should not be a reserved word like <()|&!]|:\*
-}
+  }
 
-interface BaseNode<T extends string | undefined> {
+  interface BaseNode<T extends string | undefined> {
     readonly type: T;
     readonly negated: boolean | undefined;
     readonly left: TsqueryNode | undefined;
@@ -21,44 +22,54 @@ interface BaseNode<T extends string | undefined> {
     readonly prefix: string | undefined;
     readonly quoted: boolean | undefined;
     toString(): string;
-}
+  }
 
-export interface AndNode extends BaseNode<"&"> {
+  export interface AndNode extends BaseNode<"&"> {
     readonly left: TsqueryNode;
     readonly right: TsqueryNode;
     readonly value: undefined;
     readonly prefix: undefined;
     readonly quoted: undefined;
-}
-export interface OrNode extends BaseNode<"|"> {
+  }
+
+  export interface OrNode extends BaseNode<"|"> {
     readonly left: TsqueryNode;
     readonly right: TsqueryNode;
     readonly value: undefined;
     readonly prefix: undefined;
     readonly quoted: undefined;
-}
+  }
 
-export interface FollowedByNode extends BaseNode<`<${"-" | number}>`> {
+  export interface FollowedByNode extends BaseNode<`<${"-" | number}>`> {
     readonly left: WordNode;
     readonly right: WordNode;
     readonly value: undefined;
     readonly prefix: undefined;
     readonly quoted: undefined;
-}
+  }
 
-export interface WordNode extends BaseNode<undefined> {
+  export interface WordNode extends BaseNode<undefined> {
     readonly value: string;
     readonly left: undefined;
     readonly right: undefined;
     readonly quoted: boolean;
-}
+  }
 
-export type TsqueryNode = AndNode | OrNode | FollowedByNode | WordNode;
+  export type TsqueryNode = AndNode | OrNode | FollowedByNode | WordNode;
 
-export class Tsquery {
+  export class Tsquery {
     constructor(options?: TsqueryOptions);
     parseAndStringify(str: string): string;
     parse(str: string): TsqueryNode | undefined;
+  }
 }
 
-export default function (options?: TsqueryOptions): (str: string) => string;
+declare function tsquery(options?: tsquery.TsqueryOptions): (str: string) => string;
+
+interface tsquery {
+  (options?: tsquery.TsqueryOptions): (str: string) => string;
+  Tsquery: tsquery.Tsquery;
+  Node: tsquery.TsqueryNode;
+}
+
+export = tsquery;

